### PR TITLE
Feature/search component

### PR DESCRIPTION
**Pull request by rakeshkumar-verma**

### CODE REVIEW

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml [hunk 1]
```diff
@@ -352,6 +352,19 @@
                         <jcr:content jcr:primaryType="nt:unstructured"/>
                     </policy_1587530274257>
                 </header>
+                <search jcr:primaryType="nt:unstructured">
+                    <policy_1589453975454
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml:356]: @rakeshkumar-verma : where is this policy being used ?

would need to reference same at template level

File: ui.content/src/main/content/META-INF/vault/filter.xml [hunk 2]
```diff
@@ -4,4 +4,5 @@
     <filter root="/content/hclecomm" mode="merge"/>
     <filter root="/content/dam/hclecomm" mode="merge"/>
     <filter root="/content/experience-fragments/hclecomm" mode="merge"/>
+	<filter root="/oak:index/hclecommLuceneIndex" />
```
Comment (rishabh-behl) [inline ui.content/src/main/content/META-INF/vault/filter.xml]: @rakeshkumar-verma : add mode merge here

Comment (rishabh-behl) [reply ui.content/src/main/content/META-INF/vault/filter.xml]: @rakeshkumar-verma : why do we have <filter root="/oak:index/hclecommLuceneIndex" /> as part of ui.content pom ? why not add in apps ?

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        compatVersion="{Long}2"
+        evaluatePathRestrictions="{Boolean}true"
+        excludedPaths="[/var,/etc/replication,/etc/workflow/instances,/jcr:system]"
+        fulltextEnabled="{Boolean}true"
+        includedPaths="/content/hclecomm"
+        indexPath="/oak:index/hclecommLuceneIndex"
+        name="hclecommLuceneIndex"
+        queryPaths="/content/hclecomm"
+        reindex="{Boolean}false"
+        reindexCount="{Long}5"
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/_oak_index/.content.xml]: @rakeshkumar-verma : set it to 1

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        jcr:primaryType="oak:QueryIndexDefinition"
+        async="async"
+        compatVersion="{Long}2"
+        evaluatePathRestrictions="{Boolean}true"
+        excludedPaths="[/var,/etc/replication,/etc/workflow/instances,/jcr:system]"
+        fulltextEnabled="{Boolean}true"
+        includedPaths="/content/hclecomm"
+        indexPath="/oak:index/hclecommLuceneIndex"
+        name="hclecommLuceneIndex"
+        queryPaths="/content/hclecomm"
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/_oak_index/.content.xml:27]: @rakeshkumar-verma : add isRegexp - false property

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                <include6
+                    jcr:primaryType="nt:unstructured"
+                    path="*/*/*/*/*/*/*"/>
+                <include7
+                    jcr:primaryType="nt:unstructured"
+                    path="*/*/*/*/*/*/*/*"/>
+            </cq:PageContent>
+        </aggregates>
+        <indexRules jcr:primaryType="nt:unstructured">
+            <nt:base
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/_oak_index/.content.xml]: @rakeshkumar-verma : should this not be cq:Page ??

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                jcr:primaryType="nt:unstructured"
+                includePropertyTypes="all">
+                <properties jcr:primaryType="nt:unstructured">
+                    <prop0
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/_oak_index/.content.xml]: @rakeshkumar-verma : whys is this prop0 , we are indexing jcr:Title here

**Review by rishabh-behl — COMMENTED** [review]
@rakeshkumar-verma : pl look at comments

UI is also not correct

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml (hunk 1 above)
Comment (rakeshkumar-verma) [reply ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml:356]: @rishabh-behl  updated template specific policies.

File: ui.content/src/main/content/META-INF/vault/filter.xml (hunk 2 above)
Comment (rakeshkumar-verma) [reply ui.content/src/main/content/META-INF/vault/filter.xml]: @rishabh-behl  merge mode updated in filter. Build was breaking while including oak:index in ui.apps due to root filter root's ancestor.

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml (hunk 3 above)
Comment (rakeshkumar-verma) [reply ui.content/src/main/content/jcr_root/_oak_index/.content.xml]: @rishabh-behl  updated.

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml (hunk 4 above)
Comment (rakeshkumar-verma) [reply ui.content/src/main/content/jcr_root/_oak_index/.content.xml:27]: @rishabh-behl  done.

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml (hunk 5 above)
Comment (rakeshkumar-verma) [reply ui.content/src/main/content/jcr_root/_oak_index/.content.xml]: @Rajamouli-hcl  updated to cq:page

File: ui.content/src/main/content/jcr_root/_oak_index/.content.xml (hunk 6 above)
Comment (rakeshkumar-verma) [reply ui.content/src/main/content/jcr_root/_oak_index/.content.xml]: @rishabh-behl  updated to jct:title

**Review by rishabh-behl — APPROVED** [review]
LGTM

**Comment by rishabh-behl** [comment]
approve

**Pull request merged**